src/main/java/org/springframework/data/tarantool/core/TarantoolTemplate.java

```diff
@@ -114,44 +104,16 @@ public <T> T callForObject(String functionName, List<?> parameters, Class<T> ent
         );
     }
 
-    private MessagePackMapper getDefaultComplexTypesMapper() {
-        // FIXME: https://github.com/tarantool/cartridge-java/issues/166
-        // return tarantoolClient.getConfig().getMessagePackMapper().copy();
-        return DefaultMessagePackMapperFactory
-                .getInstance()
-                .defaultComplexTypesMapper();
-    }
-
-    private void registerTupleResultMapper(MessagePackMapper customMapper,
-                                           Optional<TarantoolSpaceMetadata> spaceMetadata) {
-        customMapper.registerValueConverter(
-                ValueType.ARRAY,
-                TarantoolTuple.class,
-                new TarantoolAutoResultConverter<>(
-                        new TarantoolTupleConverter(customMapper,
-                                spaceMetadata.orElse(null)),
-                        spaceMetadata.orElse(null)));
-        customMapper.registerValueConverter(
-                ValueType.MAP,
-                TarantoolTuple.class,
-                new TarantoolAutoResultConverter<>(
-                        new TarantoolTupleConverter(customMapper,
-                                spaceMetadata.orElse(null)),
-                        spaceMetadata.orElse(null)));
-    }
-
     private <T> CallResultMapper<T, SingleValueCallResult<T>>
-    withDefaultSingleValueMapper(MessagePackMapper customMapper,
-                                 Class<T> entityClass) {
```

Comment:
Seems that `entityClass` is now missing. Aren't there any compiler warnings visible?

Reply:
No, it's purpose is broader. It allows the Java compiler to have a certain assumption about the expression type at compile time. That's the limitation of Java, but it helps t avoid some runtime errors earlier, or you will get a clearer error message. Also it saves you from further spurious unchecked casts at different places that just make the readability worse.

Reply:
You're right. It's w/a to get all types by one Mapper

```java
<T> CompletableFuture<T> callForSingleResult(
        String functionName,
        List<?> arguments,
        CallResultMapper<T, SingleValueCallResult<T>> resultMapper)
        throws TarantoolClientException;
```

We have a generic `T` in all methods with callForSingleResult in the result mapper. It forces us to make a specific class that we should obtain, only this one. That obliges us to have a container for results like TarantoolResult. And I just used Object as a generic type for further values except for the first layer which is SingleValueCallResult

Reply:
I don't like that we have raw Objects here. One of the targets of our driver API is to avoid any casts and raw Object usage whenever possible. And we shouldn't try to minimize the conversion code here in the SpringData module, if that worsens the user-facing API.